x/group/keeper/keeper_test.go

```diff
@@ -2355,6 +2417,199 @@ func (s *TestSuite) TestExecProposal() {
 	}
 }
 
+func (s *TestSuite) TestLeaveGroup() {
+	addrs := simapp.AddTestAddrsIncremental(s.app, s.sdkCtx, 7, sdk.NewInt(3000000))
+	admin := addrs[0]
+	member1 := addrs[1]
+	member2 := addrs[2]
+	member3 := addrs[3]
+	member4 := addrs[4]
+	admin2 := addrs[5]
+	admin3 := addrs[6]
+
+	require := s.Require()
+	res, err := s.keeper.CreateGroup(s.ctx, &group.MsgCreateGroup{
+		Admin: admin.String(),
+		Members: []group.Member{
+			{
+				Address:  member1.String(),
+				Weight:   "1",
+				Metadata: "metadata",
+				AddedAt:  s.sdkCtx.BlockTime(),
+			},
+			{
+				Address:  member2.String(),
+				Weight:   "2",
+				Metadata: "metadata",
+				AddedAt:  s.sdkCtx.BlockTime(),
+			},
+			{
+				Address:  member3.String(),
+				Weight:   "3",
+				Metadata: "metadata",
+				AddedAt:  s.sdkCtx.BlockTime(),
+			},
+		},
+	})
+	require.NoError(err)
+	require.NotNil(res)
+
+	res1, err := s.keeper.CreateGroup(s.ctx, &group.MsgCreateGroup{
+		Admin: admin2.String(),
+		Members: []group.Member{
+			{
+				Address:  member1.String(),
+				Weight:   "1",
+				Metadata: "metadata",
+				AddedAt:  s.sdkCtx.BlockTime(),
+			},
+		},
+	})
+	require.NoError(err)
+	require.NotNil(res1)
+
+	res2, err := s.keeper.CreateGroup(s.ctx, &group.MsgCreateGroup{
+		Admin: admin3.String(),
+		Members: []group.Member{
+			{
+				Address:  member1.String(),
+				Weight:   "1",
+				Metadata: "metadata",
+				AddedAt:  s.sdkCtx.BlockTime(),
+			},
+			{
+				Address:  member2.String(),
+				Weight:   "2",
+				Metadata: "metadata",
+				AddedAt:  s.sdkCtx.BlockTime(),
+			},
+		},
+	})
+	require.NoError(err)
+	require.NotNil(res2)
+
+	groupPolicy := &group.MsgCreateGroupPolicy{
+		Admin:    admin.String(),
+		GroupId:  res.GroupId,
+		Metadata: "metadata",
+	}
+	policy := group.NewThresholdDecisionPolicy(
+		"3",
+		time.Hour,
+		time.Hour,
+	)
+	require.NoError(groupPolicy.SetDecisionPolicy(policy))
+	require.NoError(err)
+
+	policyRes, err := s.keeper.CreateGroupPolicy(s.ctx, groupPolicy)
+	require.NoError(err)
+	require.NotNil(policyRes)
+
+	groupPolicy = &group.MsgCreateGroupPolicy{
+		Admin:    admin3.String(),
+		GroupId:  res2.GroupId,
+		Metadata: "metadata",
+	}
+	pPolicy := &group.PercentageDecisionPolicy{
+		Percentage: "0.5",
+		Windows:    &group.DecisionPolicyWindows{VotingPeriod: time.Hour},
+	}
+	require.NoError(groupPolicy.SetDecisionPolicy(pPolicy))
+	require.NoError(err)
+
+	policyRes1, err := s.keeper.CreateGroupPolicy(s.ctx, groupPolicy)
+	require.NoError(err)
+	require.NotNil(policyRes1)
```

Comment:
you could update the existing `createGroupAndGroupPolicy` function to take more parameters (group/member/policy info) and just use it here instead